Framework/CurveFitting/src/TableWorkspaceDomainCreator.cpp

```diff
@@ -0,0 +1,772 @@
+      // the Openning one should go first (be "smaller").
+      // This way the procedure of joinOverlappingRanges will join
+      // the ranges that meet at these points.
+      return this->kind == Openning;
+    }
+    return this->value < point.value;
+  }
+};
+
+/// Find any overlapping ranges in a vector and join them.
+/// @param[in,out] exclude :: A vector with the ranges some of which may
+/// overlap. On output all overlapping ranges are joined and the vector contains
+/// increasing series of doubles (an even number of them).
+void joinOverlappingRanges(std::vector<double> &exclude) {
+  if (exclude.empty()) {
+    return;
+  }
+  // The situation here is similar to matching brackets in an expression.
+  // If we sort all the points in the input vector remembering their kind
+  // then a separate exclusion region starts with the first openning bracket
+  // and ends with the matching closing bracket. All brackets (points) inside
+  // them can be dropped.
+
+  // Wrap the points into helper struct RangePoint
+  std::vector<RangePoint> points;
+  points.reserve(exclude.size());
+  for (auto point = exclude.begin(); point != exclude.end(); point += 2) {
+    points.push_back(RangePoint{RangePoint::Openning, *point});
+    points.push_back(RangePoint{RangePoint::Closing, *(point + 1)});
+  }
+  // Sort the points according to the operator defined in RangePoint.
+  std::sort(points.begin(), points.end());
+
+  // Clear the argument vector.
+  exclude.clear();
+  // Start the level counter which shows the number of unmatched openning
+  // brackets.
+  size_t level(0);
+  for (auto &point : points) {
+    if (point.kind == RangePoint::Openning) {
+      if (level == 0) {
+        // First openning bracket starts a new exclusion range.
+        exclude.push_back(point.value);
+      }
+      // Each openning bracket increases the level
+      ++level;
+    } else {
+      if (level == 1) {
+        // The bracket that makes level 0 is an end of a range.
+        exclude.push_back(point.value);
+      }
+      // Each closing bracket decreases the level
+      --level;
+    }
+  }
+}
+
+bool greaterIsLess(double x1, double x2) { return x1 > x2; }
+} // namespace
+
+using namespace Kernel;
+using API::MatrixWorkspace;
+using API::Workspace;
+
+/**
+ * Constructor.
+ * @param fit :: Property manager with properties defining the domain to be
+ * created
+ * @param workspacePropertyName :: Name of the workspace property.
+ * @param domainType :: Type of the domain: Simple, Sequential, or Parallel.
+ */
+TableWorkspaceDomainCreator::TableWorkspaceDomainCreator(
+    Kernel::IPropertyManager *fit, const std::string &workspacePropertyName,
+    TableWorkspaceDomainCreator::DomainType domainType)
+    : IDomainCreator(fit, std::vector<std::string>(1, workspacePropertyName),
+                     domainType),
+      m_startX(EMPTY_DBL()), m_endX(EMPTY_DBL()), m_maxSize(0),
+      m_xColumnIndex(0), m_yColumnIndex(1), m_errorColumnIndex(2) {
```

Comment:
I have a suggestion regarding the column selection.

Instead of using the first 3 columns could we by default first check if any columns have been set as "plot type" X, Y, E and use the first of each of those that is found. In this case no columns are selected I'd suggested using your current case of the 3 columns but only if the size is exactly 3 else its ambiguous and I think better in that scenario to just say they need to specify the columns explicitly.

What do you think?

Reply:
This is a good idea, makes more sense than how it currently is.